src/packageurl/__init__.py

```diff
@@ -374,6 +374,36 @@ def from_string(cls, purl):
             raise ValueError(msg)
 
         path = path.lstrip("/")
+
+        # NPM purl have a namespace in the path
+        # and the namespace in an npm purl is
+        # different from others because it starts with `@`
+        # so we need to handle this case separately
+        if type == "npm" and path.startswith("@"):
+            namespace = ""
+            name = ""
+            namespace, sep, name_version = path.partition("/")
```

Comment:
```suggestion
            namespace, sep, path = path.partition("/")
```
and then just do not special case anymore and the new code added below